pkg/caveats/compile.go

```diff
@@ -48,6 +55,15 @@ type CompilationErrors struct {
 	issues *cel.Issues
 }
 
+func CompileCaveatWithName(env *Environment, exprString, name string) (*CompiledCaveat, error) {
+	c, err := CompileCaveat(env, exprString)
+	if err != nil {
+		return nil, err
+	}
+	c.name = name
```

Comment:
We should probably store the name in the compiled caveat too

Reply:
Yeah I didn't know how to proceed with the `pkg/caveats` bits so left it intentionally incomplete. Will look into that 👍🏻

Reply:
addressed in https://github.com/authzed/spicedb/pull/807/commits/015507d6de0d01742c7e1ccb68e1093ff5979ed0